celery_tasks/tasks.py

```diff
@@ -0,0 +1,21 @@
+import pendulum
+
+from django.db.models import Sum
+
+from api.models import Salary, SalaryCache
+from .celery_app import app
+
+
+@app.task
+def add_to_salary_cached(id):
+    qs = Salary.objects.filter(company_employee__employee__id=id) \
```

Comment:
1. The amount to recalculate is for the `calendary year corresponding to the given date`, not for a `year back from now`
1. if there are missing months (e.g. it's Jun 2019 and Jul-Dec have not yet come), then you have to recalculate for Jan-Jun of 2019. But in general calculating for the whole 2019.